Server/src/main/java/net/simon987/server/assembly/CPU.java

```diff
@@ -466,6 +468,14 @@ public CpuHardware getHardware(int address) {
         return attachedHardware.get(address);
 
     }
-
-
+    /**
+     * Sets the IP to 0x0200 + Offset and pushes flags then the old IP
+     * 
+     */
+    public void Interrupt(int hw, int offset, boolean retry) {
+        Instruction push = instructionSet.get("push");
+        push.execute(status.toByte(), status);
+        push.execute(ip, status);
+        this.setIp((char)(0x0200 + offset*2 + 0x0080*hw));
```

Comment:
`0x200` Shouldn't be hardcoded, as the code origin can be changed with the ORG directive. (Should be CPU.codeSegmentOffset)